src/viewer.js

```diff
@@ -48,22 +48,34 @@ async function startServer(bundleStats, opts) {
 
   if (!chartData) return;
 
-  const app = express();
-  app.use(express.static(`${projectRoot}/public`));
-
-  app.get('/', (req, res) => {
-    res.writeHead(200, {'Content-Type': 'text/html'});
-    const html = renderViewer({
-      mode: 'server',
-      title: resolveTitle(reportTitle),
-      chartData,
-      defaultSizes,
-      enableWebSocket: true
-    });
-    return res.end(html);
+  const sirvMiddleware = sirv(`${projectRoot}/public`, {
+    dev: true
   });
 
-  const server = http.createServer(app);
+  const server = http.createServer((req, res) => {
+    if (req.method === 'GET' && req.url === '/') {
+      const html = renderViewer({
+        mode: 'server',
+        title: resolveTitle(reportTitle),
+        chartData,
+        defaultSizes,
+        enableWebSocket: true
+      });
+      res.writeHead(200, {'Content-Type': 'text/html'});
+      res.end(html);
+    } else {
+      sirvMiddleware(req, res, err => {
+        if (err) {
+          console.error(err.stack || err.toString());
+          res.writeHead(500);
+          res.end();
+        } else {
+          res.writeHead(404);
```

Comment:
Why 404?

Reply:
Yeah, callback is not necessary. This is default behaviour. Removed.